filament/src/details/SkinningBuffer.cpp

```diff
@@ -162,5 +164,101 @@ void FSkinningBuffer::setBones(FEngine& engine, Handle<backend::HwBufferObject>
             offset * sizeof(PerRenderableBoneUib::BoneData));
 }
 
+// This value is limited by ES3.0, ES3.0 only guarantees 2048.
+// When you change this value, you must change MAX_SKINNING_BUFFER_WIDTH at getters.vs
+constexpr size_t MAX_SKINNING_BUFFER_WIDTH = 2048;
+
+static inline size_t getWidth(size_t pairCount) noexcept {
+    return std::min(pairCount, MAX_SKINNING_BUFFER_WIDTH);
+}
+
+static inline size_t getHeight(size_t pairCount) noexcept {
+    return (pairCount + MAX_SKINNING_BUFFER_WIDTH) / MAX_SKINNING_BUFFER_WIDTH;
+}
+
+inline size_t getSize(size_t pairCount) noexcept {
```

Comment:
these names are a little too generic, either move them to a class, or use something more descriptive.

Reply:
ok, renamed